File: 2015/day01/main.rb
File.open("input.txt", "r") do |file|
  floor = 0
  basementAfter = 0
  searchingBasement = true

  while c = file.getc do
    floor += if c == '(' then 1 else -1 end
    if searchingBasement
      basementAfter += 1
      searchingBasement = floor != -1;
    end
  end

  puts "Part 1: #{floor}\nPart 2: #{basementAfter}\n"

end

File: 2015/day01/main.hs

walk :: String -> Int
walk = sum . map move

basementAfter :: String -> Int
basementAfter = (+1) . length . takeWhile (/= -1) . scanl1 (+) . map move

move :: Char -> Int
move '(' = 1
move ')' = -1

main :: IO()
main = main_p2 where
  main_p1 = main' walk
  main_p2 = main' basementAfter
  main' f = readFile "input.txt" >>= print . f
